File: src/main/java/io/metjka/vortex/ui/blocks/MathOutputBlock.java
package io.metjka.vortex.ui.blocks;

import com.google.common.collect.ImmutableList;
import io.metjka.vortex.ui.ToplevelPane;
import io.metjka.vortex.ui.connections.OutputAnchor;
import io.metjka.vortex.ui.connections.Target;
import javafx.fxml.FXML;
import javafx.scene.control.Label;
import javafx.scene.layout.Pane;

import java.util.List;
import java.util.Optional;

public class MathOutputBlock extends Block implements Target {

    protected InputAnchor inputAnchor;

    @FXML
    private Pane inputSpace;

    @FXML
    private Label printValue;

    /**
     * @param pane The pane this block belongs to.
     */
    public MathOutputBlock(ToplevelPane pane) {
        super(pane, "InvertBlock");
        loadFXML("MathOutputBlock");

        inputAnchor = new InputAnchor(this, Type.NUMBER);
        inputSpace.getChildren().add(0, inputAnchor);
    }

    @Override
    public ConnectionAnchor getAssociatedAnchor() {
        return inputAnchor;
    }

    @Override
    public List<InputAnchor> getAllInputs() {
        return ImmutableList.of(inputAnchor);
    }

    @Override
    public List<OutputAnchor> getAllOutputs() {
        return ImmutableList.of();
    }

    @Override
    public void update() {
        inputAnchor.invalidateVisualState();
        if (inputAnchor.getOppositeAnchor().isPresent()) {

            OutputAnchor outputAnchor = inputAnchor.getOppositeAnchor().get();

            Block block = outputAnchor.getBlock();
            int position = outputAnchor.getPosition();

            printValue.setText(block.getValueFromBlock(position).toString());


        }


    }


    @Override
    public Optional<Block> getNewCopy() {
        return Optional.of(new MathOutputBlock(this.getToplevel()));
    }

}

File: src/main/java/io/metjka/vortex/ui/blocks/MathOutputBlock.kt
package io.metjka.vortex.ui.blocks;

import io.metjka.vortex.ui.TopLevelPane
import io.metjka.vortex.ui.Type
import io.metjka.vortex.ui.connections.InputAnchor
import io.metjka.vortex.ui.connections.OutputAnchor
import javafx.fxml.FXML
import javafx.scene.control.Label
import javafx.scene.layout.Pane
import java.util.*

class MathOutputBlock(topLevelPane: TopLevelPane) : Block(topLevelPane, MathOutputBlock::class.simpleName) {

    val inputAnchor: InputAnchor<Int> = InputAnchor<Int>(this, Type.NUMBER);

    @FXML
    val inputSpace: Pane? = null

    @FXML
    val printValue: Label? = null

    /**
     * @param pane The pane this block belongs to.
     */
   init {
        inputSpace?.getChildren()?.add(0, inputAnchor);

    }

    override fun getAllInputs():List<InputAnchor<*>> {
        return listOf(inputAnchor)
    }

    override fun getAllOutputs():List<OutputAnchor<*>> {
        return emptyList()
    }

    override fun update() {
        inputAnchor.invalidateVisualState()

        val property: Int = inputAnchor.getOppositeAnchor().get().property?.value!!
        printValue?.text = property.toString()

    }

    override fun getNewCopy(): Optional<Block> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }




}
